File: src/main/java/generator/TemplateGeneratorPluginsAction.java
package generator;

import com.intellij.database.model.ObjectKind;
import com.intellij.openapi.actionSystem.*;
import generator.ui.dialog.GenerateConfigDialog;
import generator.util.DasUtil;
import org.jetbrains.annotations.NotNull;

import java.util.Set;

public class TemplateGeneratorPluginsAction extends AnAction {

    @Override
    public @NotNull ActionUpdateThread getActionUpdateThread() {
        return super.getActionUpdateThread();
    }

    private static final Set<ObjectKind> SUPPORTED_KINDS = Set.of(
            ObjectKind.TABLE,
            ObjectKind.VIEW,
            ObjectKind.SCHEMA,
            ObjectKind.DATABASE
    );
    @Override
    public void update(@NotNull AnActionEvent e) {
        var dasObjectStream = DasUtil.extractDatabaseDas(e.getDataContext());

        var b = dasObjectStream.anyMatch(x -> SUPPORTED_KINDS.contains(x.getKind()));
        e.getPresentation().setEnabledAndVisible(b);
        super.update(e);
    }

    @Override
    public void actionPerformed(AnActionEvent e) {
        var tableGenerator = new GenerateConfigDialog(e);

        tableGenerator.setTitle("TemplateGenerator");
        tableGenerator.setSize(800, 600);

        tableGenerator.show();
    }
}

File: src/main/java/generator/TemplateGeneratorPluginsAction.kt
package generator

import com.intellij.database.model.DasObject
import com.intellij.database.model.ObjectKind
import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.ui.Messages
import generator.ui.dialog.GenerateConfigDialog
import generator.util.DasUtil
import java.io.PrintWriter
import java.io.StringWriter



class TemplateGeneratorPluginsAction : AnAction() {

    override fun getActionUpdateThread(): ActionUpdateThread {
        return super.getActionUpdateThread()
    }

    override fun update(e: AnActionEvent) {
        val dasObjectStream = DasUtil.extractDatabaseDas(e.dataContext)

        val b = dasObjectStream.anyMatch { x: DasObject -> SUPPORTED_KINDS.contains(x.kind) }
        e.presentation.isEnabledAndVisible = b
        super.update(e)
    }

    override fun actionPerformed(e: AnActionEvent) {
        try {

        val tableGenerator = GenerateConfigDialog(e)

        tableGenerator.title = "TemplateGenerator"
        tableGenerator.setSize(800, 600)

        tableGenerator.show()
        } catch (e: Exception) {
            val sw = StringWriter()
            val pw = PrintWriter(sw)
            e.printStackTrace(pw) // 将堆栈信息打印到PrintWriter
            val fullStackTrace = sw.toString() // 获取完整的堆栈信息字符串

            Messages.showErrorDialog(e.message + "\n" + fullStackTrace, "Error")
        }
    }

    companion object {
        private val SUPPORTED_KINDS: Set<ObjectKind> = mutableSetOf(
            ObjectKind.TABLE,
            ObjectKind.VIEW,
            ObjectKind.SCHEMA,
            ObjectKind.DATABASE
        )
    }
}
